Bori patron deity

Bannana-Puncakes opened the pull request:
Edited the pick main deity decision in minor_decisions to include bori and call the corresponding event. Written events for choosing the deities as specified in the spreadsheet. Implemented the localisations and modifiers from the spreadsheet.

Commit:
Bori Patron Deity Events

Edited the pick main deity event to include bori. Written bori specific events for choosing a patron deity and added associated localisations and modifiers.

Commit:
Bori Patron Deity decision and events

Commit:
Bori Patron Deity decision and events

Actually right this time...

Commit:
Fixes from game testing

Commit:
Final commit?!

Final set of fixes

Commit:
Added Bori main deity decision and events

Ammended the Choose Main Deity decision to allow for bori characters. Created new events to choose from the six options and added appropriate localisations and modifiers

faiuwle commented on AfricanReligions/common/event_modifiers/afr_event_modifiers.txt:
```diff
@@ -31,4 +31,40 @@ secret_amazigh_reformed_community = {
+
+follower_of_yan_riga = {
+	learning = +2
+	martial_penalty = -1
```
This should just be `martial = -1`, and the same for the others.

faiuwle commented on AfricanReligions/common/event_modifiers/afr_event_modifiers.txt:
```diff
@@ -31,4 +31,40 @@ secret_amazigh_reformed_community = {
+}
+
+follower_of_yan_riga = {
+	learning = +2
```
No + needed here (or on the others).

faiuwle commented on AfricanReligions/events/afr_misc_events.txt:
```diff
@@ -235,4 +235,219 @@ character_event = {
+
+    immediate = {
+        if = {
+            limit = { if_playable = yes }
```
Typo - `is_playable`.

faiuwle commented on AfricanReligions/events/afr_misc_events.txt:
```diff
@@ -235,4 +235,219 @@ character_event = {
+                    prisoner = no
+                    religion = root
+                    age = 16
+                    has_minor_deity_trigger = no
```
If you want this to work, you need to modify this trigger so that it returns true if the character has one of our modifiers.

Bannana-Puncakes replied:
I'm not actually sure what this does I copied the event structure of the Slavic pick patron deity. There was also an identical event just without the immediate bit so idk... could likely just remove it?

faiuwle replied:
No, you need it here.  It's a scripted trigger, they are defined in common/scripted_triggers, essentially it is a condition block that can be reused like a function.  But this one only checks for vanilla patron deities, so you'll have to add ours to it, it is used here to fire the patron deity event for all of the player's courtiers that don't have one - the condition block limits the scope to just courtiers that don't have a patron deity.

faiuwle commented on AfricanReligions/events/afr_misc_events.txt:
```diff
@@ -235,4 +235,219 @@ character_event = {
+    }
+
+    option = {
+        name = EVTOPTA_AFRREL00501
```
You can name the localization keys anything, but I think in general we should name them so that we know what they're for later.  So, the first option name of event 00500 would be EVTOPTA_AFRREL00500, and the next one would be EVTOPTB_AFRREL00500, etc.  The ones ending in AFRREL00501 would then be fore event 00501 if it exists, and so forth.

Bannana-Puncakes replied:
What do you mean because I think this is what I've done already?

faiuwle replied:
Your key says EVTOPTA_AFRREL00501, which indicates that it's the first option on event 00501, but it's actually the first option on event 00500.  The second option is EVTOPTA_AFRREL00502, which indicates that it's the first option on event 00502, but actually it's the second option on event 00500.

faiuwle commented on AfricanReligions/events/afr_misc_events.txt:
```diff
@@ -235,4 +235,219 @@ character_event = {
+        add_character_modifier = {
+            name = follower_of_yan_riga
+            duration = -1
+        }
```
Let's only assign the modifier in the final event, and let people go back to this one, that way people can read all the localization if they want to.  Instead, here you should add a tooltip that indicates the choice of this option isn't final, like "You will consider the Yan Riga" or something like that.

faiuwle commented on AfricanReligions/localisation/afr_event_texts.csv:
```diff
@@ -78,4 +78,28 @@ EVTDESC_MNM_3227_AFRICAN;As I step up to the [Root.GetReligiousCeremonyLocation]
+EVTDESC_AFRREL00500;In the ethereal realm of Jan Gari, the Bori spirits exist in a transcendental state incomprehensible to the mortals. If I make special contact with one of the great bori clans and enter into communion with their king, it is possible that the spirits will grant me their blessing.;;;;;;;;;;;;;x
+EVTOPTA_AFRREL00501;Time to chose.;;;;;;;;;;;;;x
+EVTOPTA_AFRREL00502;Maybe another time.;;;;;;;;;;;;;x
+EVTDESC_AFRREL00510;While I make an effort so that my soul comes in contact with the Jan Gari plan, I face the decision of which spirit I should ask for favor. There are six major categories of spirits each with their own hierarchy and a powerful spiritual king who rules over them. Which of these spiritual courts will I pay homage to?;;;;;;;;;;;;;x
```
I think "plan" was meant to be "plane".

faiuwle commented on AfricanReligions/localisation/afr_event_texts.csv:
```diff
@@ -78,4 +78,28 @@ EVTDESC_MNM_3227_AFRICAN;As I step up to the [Root.GetReligiousCeremonyLocation]
+EVTOPTA_AFRREL00512;Yan Riga, the city spirits.;;;;;;;;;;;;;x
+EVTOPTA_AFRREL00513;Babbaku, the rural spirits.;;;;;;;;;;;;;x
+EVTOPTA_AFRREL00514;Tan Garki, the warrior spirits.;;;;;;;;;;;;;x
+EVTOPTA_AFRREL00515;More options.;;;;;;;;;;;;;x
```
How about something more immersive, like "There are other spirits to choose..." or "I had something else in mind."

faiuwle commented on AfricanReligions/localisation/afr_event_texts.csv:
```diff
@@ -78,4 +78,28 @@ EVTDESC_MNM_3227_AFRICAN;As I step up to the [Root.GetReligiousCeremonyLocation]
+EVTOPTA_AFRREL00513;Babbaku, the rural spirits.;;;;;;;;;;;;;x
+EVTOPTA_AFRREL00514;Tan Garki, the warrior spirits.;;;;;;;;;;;;;x
+EVTOPTA_AFRREL00515;More options.;;;;;;;;;;;;;x
+EVTOPTA_AFRREL00516;Samari, the young femenine spirits.;;;;;;;;;;;;;x
```
"feminine"

faiuwle commented on AfricanReligions/localisation/afr_event_texts.csv:
```diff
@@ -78,4 +78,28 @@ EVTDESC_MNM_3227_AFRICAN;As I step up to the [Root.GetReligiousCeremonyLocation]
+EVTOPTA_AFRREL00530;With the blessing of this spiritual court I will also know more about the land, making me a better administrator.;;;;;;;;;;;;;x
+EVTDESC_AFRREL00540;The Tan Garki, that is, the sons of the shield, are the great warriors among the spirits. Many brave fighters seek their protection, but the true custodians of their blessings are the generals and strategists. The head of this spiritual army is Garki Baba, the great shield, protector of those who march to the battlefield.;;;;;;;;;;;;;x
+EVTOPTA_AFRREL00540;With the blessing of this spiritual court I will be able to guide my army with ease.;;;;;;;;;;;;;x
+EVTDESC_AFRREL00550;Samari, the spirits of youth but also of femininity. Interestingly they are led by Ba-Gudu, the one who does not run, who is neither female nor young. It is a court full of curious talents such as giving good or bad luck to gamblers, persuade a man to desire his brother's wife, or making the breasts of young girls irritate by awakening her desires.;;;;;;;;;;;;;x
```
"persuading"

faiuwle commented on AfricanReligions/localisation/afr_event_texts.csv:
```diff
@@ -78,4 +78,28 @@ EVTDESC_MNM_3227_AFRICAN;As I step up to the [Root.GetReligiousCeremonyLocation]
+EVTOPTA_AFRREL00540;With the blessing of this spiritual court I will be able to guide my army with ease.;;;;;;;;;;;;;x
+EVTDESC_AFRREL00550;Samari, the spirits of youth but also of femininity. Interestingly they are led by Ba-Gudu, the one who does not run, who is neither female nor young. It is a court full of curious talents such as giving good or bad luck to gamblers, persuade a man to desire his brother's wife, or making the breasts of young girls irritate by awakening her desires.;;;;;;;;;;;;;x
+EVTOPTA_AFRREL00550;The blessing of such a diverse court will surely make me a people person.;;;;;;;;;;;;;x
+EVTDESC_AFRREL00560;The Yayan zanzanna, the children of the plague, are kid spirits with dominion over any disease, spreading smallpox and other evils among mortals. They are led by Nana Mariyamu, the naughty little queen who likes to give headaches and irritated eyes to those who are mean to her. The whole court is playful and likes to clap for no reason, after all they are just children.;;;;;;; ...
```
"child" sounds better than "kid" here.

faiuwle commented on AfricanReligions/localisation/afr_general_localization.csv:
```diff
@@ -26,7 +26,7 @@ GOD_TIURAKH;Tiurakh;;;;;;;;;;;;;x
 GOD_THE_EVIL_PANGOOLS;The evil pangools;;;;;;;;;;;;;x
 GOD_TANIT;Tanit;;;;;;;;;;;;;x
-GOD_ANTI;�nti;;;;;;;;;;;;;x
+GOD_ANTI;Änti;;;;;;;;;;;;;x
```
It looks like you've converted these files into UTF-8.  You'll have to convert them back in Windows-1252/ANSI and then make sure your editor is set to that encoding.

faiuwle commented on AfricanReligions/localisation/afr_general_localization.csv:
```diff
@@ -324,4 +324,11 @@ secret_religious_society_amazigh_reformed_rank_2_female;Adherent;;;;;;;;;;;;;x
-secret_religious_society_amazigh_reformed_rank_4_female;Herald;;;;;;;;;;;;;x
\ No newline at end of file
+secret_religious_society_amazigh_reformed_rank_4_female;Herald;;;;;;;;;;;;;x
+<<<<<<< Updated upstream
```
Looks like something git inserted?

faiuwle commented on AfricanReligions/localisation/afr_general_localization.csv:
```diff
@@ -324,4 +324,11 @@ secret_religious_society_amazigh_reformed_rank_2_female;Adherent;;;;;;;;;;;;;x
\ No newline at end of file
+secret_religious_society_amazigh_reformed_rank_4_female;Herald;;;;;;;;;;;;;x
+<<<<<<< Updated upstream
+follower_of_yan_riga; Follower of Yan Riga;;;;;;;;;;;;;x
```
I don't think we want a space at the beginning.

faiuwle commented on AfricanReligions/events/afr_misc_events.txt:
```diff
@@ -235,4 +235,219 @@ character_event = {
+    is_triggered_only = yes
+
+    option = {
+        name = EVTOPTA_AFRREL00520
```
All of these option texts except for the Samari and the Yayan Zanzanna one are too long for the box.  You need to ask TheDude to write shorter ones.

faiuwle commented on AfricanReligions/events/afr_misc_events.txt:
```diff
@@ -235,4 +235,219 @@ character_event = {
+    option = {
+        name = EVTOPTA_AFRREL00502
+        piety = 25
+        remove_character_modifier = picked_main_deity_timer
```
This line needs to be in a `hidden_tooltip` or else it will be displayed to the player.

faiuwle reviewed (changes requested):
So, reusing the decision works well, but we'd ideally like to keep the special localization of the decision name and description for Bori characters.  You can do this by overriding the decision localization (put it in 000_vanilla_overrides.csv) with a customizable localization, if you need help figuring out how that works let me know.

faiuwle commented:
Sorry, one more thing, you should add localizations for the modifier descriptions, which are going to be `<modifier_id>_desc`.

Commit:
PR requested changes

Made changes to code as requrested by the PR review. Still need to add event pictures.

Commit:
Added event pictures

Added event pictures for the bori pick patron deity events and made appropriate changes to scripts.

faiuwle commented on AfricanReligions/common/event_modifiers/afr_event_modifiers.txt:
```diff
@@ -31,4 +31,40 @@ secret_amazigh_reformed_community = {
+}
+
+follower_of_babbaku = {
+	stewardship = 1
```
Should be 2 stewardship.

faiuwle commented on AfricanReligions/localisation/afr_general_localizations.csv:
```diff
@@ -324,4 +324,20 @@ secret_religious_society_amazigh_reformed_rank_2_female;Adherent;;;;;;;;;;;;;x
-secret_religious_society_amazigh_reformed_rank_4_female;Herald;;;;;;;;;;;;;x
\ No newline at end of file
+secret_religious_society_amazigh_reformed_rank_4_female;Herald;;;;;;;;;;;;;x
+follower_of_yan_riga;Follower of Yan Riga;;;;;;;;;;;;;x
```
Let's put these localizations up with the other event modifiers (secret_X_community).

faiuwle commented on AfricanReligions/localisation/afr_general_localizations.csv:
```diff
@@ -324,4 +324,20 @@ secret_religious_society_amazigh_reformed_rank_2_female;Adherent;;;;;;;;;;;;;x
+follower_of_yayan_zanzanna_desc;The Yayan Zanzanna, the children of the plague, are child spirits with dominion over any disease, spreading smallpox and other evils among mortals, Their queen is Nana Mariyamu.;;;;;;;;;;;;;x
+follower_of_tan_dowa;Follower of Tan Dowa;;;;;;;;;;;;;x
+follower_of_tan_dowa_desc;The Tan Dowa are spirits of the jungle, they do not resemble any juman but are nature in its purest state. Zaki the lion is the first among them.;;;;;;;;;;;;x
+vanilla_pick_main_deity;Pick Patron Deity;;;;;;;;;;;;x
```
Similarly, these go up with the decision localizations.

faiuwle commented on AfricanReligions/common/event_modifiers/afr_event_modifiers.txt:
```diff
@@ -31,4 +31,40 @@ secret_amazigh_reformed_community = {
+
+follower_of_tan_dowa = {
+	combat_rating = 10
+	diplomacy = 1
```
Should be -1 diplomacy.

faiuwle commented on AfricanReligions/localisation/afr_general_localizations.csv:
```diff
@@ -324,4 +324,20 @@ secret_religious_society_amazigh_reformed_rank_2_female;Adherent;;;;;;;;;;;;;x
+follower_of_tan_dowa_desc;The Tan Dowa are spirits of the jungle, they do not resemble any juman but are nature in its purest state. Zaki the lion is the first among them.;;;;;;;;;;;;x
+vanilla_pick_main_deity;Pick Patron Deity;;;;;;;;;;;;x
+vanilla_pick_main_deity_desc;There are many gods and goddesses who rule over different aspects of our life. While all of them should be honored properly, it is common for people to pick a deity whom they pray and sacrifice to more often.;;;;;;;;;;;;x
+bori_pick_main_deity;Pick Patron Deity;;;;;;;;;;;;x
```
This should be "Pick Patron Bori" or something along those lines.

faiuwle reviewed (changes requested):
Looks good!  Just a couple more changes.

faiuwle commented on AfricanReligions/localisation/afr_event_texts.csv:
```diff
@@ -78,4 +78,28 @@ EVTDESC_MNM_3227_AFRICAN;As I step up to the [Root.GetReligiousCeremonyLocation]
+EVTOPTA_AFRREL050;Time to chose.;;;;;;;;;;;;;x
+EVTOPTB_AFRREL050;Maybe another time.;;;;;;;;;;;;;x
+EVTDESC_AFRREL051;While I make an effort so that my soul comes in contact with the Jan Gari plane, I face the decision of which spirit I should ask for favor. There are six major categories of spirits each with their own hierarchy and a powerful spiritual king who rules over them. Which of these spiritual courts will I pay homage to?;;;;;;;;;;;;;x
+EVTOPTA_AFRREL051;You consider Yan Riga, the city spirits.;;;;;;;;;;;;;x
```
Sorry, another comment: this should be in the first person.

faiuwle commented on AfricanReligions/events/afr_misc_events.txt:
```diff
@@ -235,4 +235,258 @@ character_event = {
+    }
+
+    option = {
+        name = EVTOPTB_AFRREL050
```
Let's not reuse this one.  Write one about considering other bori.

Commit:
PR2 changes

Changes made just waiting on localisation information.

Commit:
Final Changes

Does what it says on the tin.

faiuwle committed:
A few localization cleanups

faiuwle committed:
Merge branch 'master' into bori-patron-deity
